### DISCUSSION TITLE
Send alerts to Slack or PagerDuty

### DISCUSSION THREAD

**Discussion by chadlangston** [Security-Onion-Solutions/securityonion#3794] [Unsupported Versions] [answered]
Is there a way to send all the alerts from the SOC Alerts page to Slack or PagerDuty? I've read through the documentation on how to configure ElastAlert to send alerts to Slack, but I'm not sure that is what I am looking for. Is there a place where I can set a global rule to send all SOC Alerts to Slack or PagerDuty?

Thank you!

**Comment by stijnpieters** [comment, marked as answer]
You could create an ElastAlert that forwards all rules to Slack. I currently have a setup that aggregates all high alerts of the last few seconds and sends them in an alert to Slack. There's also some custom formatting I did, but you could leave this out if you don't want it that way.

I wouldn't recommend forwarding EVERY alert, because low alerts usually do not mean anything. 
Forwarding every high alert instantly did not feel right for me, as some events tend to generate multiple alerts for the exact same interaction, and that way your Slack kind of just blows up. That's why I aggregate the 10 seconds. 
How often ElastAlert updates its data from Elastic is defined in its config in /opt/so/saltstack/default/salt/elastalert/files/elastalert_config.yaml.

Alert for reference:
[high_severity_slack.txt](https://github.com/Security-Onion-Solutions/securityonion/files/6279599/high_severity_slack.txt)

**Reply by chadlangston** [reply]
Cool! This is exactly what I was hoping for! I really appreciate it 🙏

**Reply by jdecker79** [reply]
I am interested in setting this up in my Security Onion System.  Would the high_severity_slack.txt file be placed in the /opt/so/rules/elastalert/playbook folder or /opt/so/rules/elastalert/ folder.  Also I assume it would need to be saved as a .yaml file?

Thanks for any help you can provide.

**Reply by stijnpieters** [reply]
/opt/so/rules/elastalert should do the trick.
it does indeed have to be a .yaml file. (couldn't upload yaml as an attachment)

**Reply by jdecker79** [reply]
Thank you for the quick response.

**Reply by jdecker79** [reply]
Other than the slack_webhook_url do I need to make custom changes to the yaml file?  I have set the event.severity: >1 just to make sure I get notifications and I have not received any even though I get many alerts during the day on the SOC Alerts.

**Reply by stijnpieters** [reply]
I think it should work by just adding your Slack webhook URL. I'm not sure anymore, but the "es_host" might also need to be changed to the hostname (or IP) of your Security Onion node
